packages/data-transport-layer/src/db/transport-db.ts

```diff
@@ -293,31 +280,46 @@ export class TransportDB {
 
     const fullTransactions = []
     for (const transaction of transactions) {
-      if (transaction.queueOrigin === 'l1') {
-        const enqueue = await this.getEnqueueByIndex(transaction.queueIndex)
-        if (enqueue === null) {
-          return null
-        }
-
-        fullTransactions.push({
-          ...transaction,
-          ...{
-            blockNumber: enqueue.blockNumber,
-            timestamp: enqueue.timestamp,
-            gasLimit: enqueue.gasLimit,
-            target: enqueue.target,
-            origin: enqueue.origin,
-            data: enqueue.data,
-          },
-        })
-      } else {
-        fullTransactions.push(transaction)
-      }
+      fullTransactions.push(await this._makeFullTransaction(transaction))
```

Comment:
Perhaps no await here and then at the bottom `return Promise.all(fullTransactions)` ?